packages/xo-server/src/xapi/index.js

```diff
@@ -565,67 +564,10 @@ export default class Xapi extends XapiBase {
     )
   }
 
-  async _deleteVm(vmOrRef, deleteDisks = true, force = false, forceDeleteDefaultTemplate = false) {
-    const $ref = typeof vmOrRef === 'string' ? vmOrRef : vmOrRef.$ref
+  async deleteVm(vmOrId, deleteDisks = true, force = false, forceDeleteDefaultTemplate = false) {
```

Comment:
Is there still some code using this?

Reply:
Why did you choose not to convert them?

Reply:
Because there is lot of change. 
IMO it would be better to do it in another PR because theses changes are not related to this bug fix directly.

Reply:
Makes sense, but why did you change some of them?

Reply:
I made some changes in `@xen-orchestra/xapi/index.js` and it seemed normal to me to also make changes in `xo-server/src/xapi/index.js`